### DISCUSSION TITLE
Overwriting internal filebeat pipelines does not work after update

### DISCUSSION THREAD

**Discussion by Rdago** [Security-Onion-Solutions/securityonion#9785] [Unsupported Versions]
Hi There, 

since we updated our SO to latest version we are facing an issue with a semi-custom ingest pipeline. 

Back then we overwrote `filebeat-7.17.4-fortinet-firewall-pipeline` by placing it in `/opt/so/saltstack/local/salt/elasticsearch/files/ingest` and this pipeline  was used to parse the data. 
After the update we noticed that placing the pipeline `filebeat-8.5.3-fortinet-firewall-pipeline` into this folder does not work anymore and the original beats-pipeline is used. 

I already created a pull request to beats repo fix the grok pattern: https://github.com/elastic/beats/pull/34479 
The problem is that the whole pipeline does not work, because it contains a space character after the syslogprio. 

Questions: 
1. Is there a way to edit/overwrite the original ingest pipeline from beats? 
2. What is the intended way to further parse documents after it was parsed by beats? (Calling another pipeline? Where to make the call?)

**Comment by cm-ops** [comment]
Have you looked at setting a final pipeline and placing your configuration there? https://www.elastic.co/guide/en/elasticsearch/reference/current/ingest.html

**Reply by Rdago** [reply]
HI @cm-ops  

thanks for making me aware of the final pipelines. I tried to use them but it looks like it is not calling the pipeline. 


What I set up: 
``` 
[root@Server ]# so-elasticsearch-index-templates-list so-fortinet
{
  "index_templates": [
    {
      "name": "so-fortinet",
      "index_template": {
        "index_patterns": [
          "so-fortinet*"
        ],
        "composed_of": [
          "agent-mappings",
          "so-fortinet-settings"
        ],
        "priority": 500
      }
    }
  ]
}


[root@Server ]# so-elasticsearch-component-templates-list so-fortinet-settings
{
  "component_templates": [
    {
      "name": "so-fortinet-settings",
      "component_template": {
        "template": {
          "settings": {
            "index": {
              "final_pipeline": "fortinet-default"
            }
          }
        }
      }
    }
  ]
}

``` 

for test purposes I just added a tag in my `fortinet-default`-Pipeline, but the documents did not change in any kind.

**Reply by cm-ops** [reply]
Can you check the pipeline stats? `so-elasticsearch-pipeline-stats $pipeline`?
